R/spark_apply.R

```diff
@@ -68,17 +68,25 @@ spark_apply <- function(x, f, names = colnames(x), memory = TRUE, ...) {
   # build the configuration definetion for each worker role
 
   # create a configuration string to initialize each worker
-  worker_config <- worker_config_serialize(list(
-    debug = isTRUE(args$debug)
-  ))
+  worker_config <- worker_config_serialize(
+    c(
+      list(
+        debug = isTRUE(args$debug)
+      ),
+      sc$config
+    )
+  )
+
+  worker_port <- spark_config_value(sc$config, "sparklyr.gateway.port", "8880")
```

Comment:
Yes, so far has worked fine. When there is only one driver, the main gateway serves all the mapping requests. For worker roles, they would start its own gateway.